### PR TITLE
Mon 13562 sonar atoi external command

### PR DESCRIPTION
**Pull request by denliA**
#REFS: MON-13562
-replace atoi/strtoul by abseil::SimpleAtoi
-robot tests : BEATOI11, BEATOI12, BEATOI13, BEATOI21, BEATOI22, BEATOI23
-unit tests : SSHOptions in connectors/ssh/tests/options.cc

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: connectors/ssh/src/orders/options.cc [hunk 1]
```diff
@@ -217,7 +218,16 @@ void options::parse(std::string const& cmdline) {
...
+        unsigned int temp;
+        if (!absl::SimpleAtoi(optarg, &temp)) {
+          throw basic_error() << "the argument '" << optarg
+                              << "' must be an unsigned short integer";
+        }
+        if (temp > 65535) {
+          throw basic_error() << "the argument '" << optarg
+                              << "' must be an integer between 0 and 65535";
+        }
+        _port = temp;
```
Comment (bouda1) [inline connectors/ssh/src/orders/options.cc:230]: Please, encapsulate this code with {}

File: engine/src/commands/commands.cc [hunk 2]
```diff
@@ -163,15 +163,18 @@ int cmd_add_comment(int cmd, time_t entry_time, char* args) {
   char* svc_description(nullptr);
   char* user(nullptr);
   char* comment_data(nullptr);
-  int persistent(0);
+  bool persistent{false};
   uint64_t service_id = 0;
+  char* command_name(nullptr);
```
Comment (bouda1) [inline engine/src/commands/commands.cc:168]: no need to initialize it. You make the work just after

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: tests/broker-engine/external-commands.robot [hunk 3]
```diff
@@ -2309,3 +2309,123 @@ BEEXTCMD_REVERSE_GRPC4
...
+	Broker Config Log	central	sql	debug
+	${start}=	Get Current Date
+	Start Broker
+	Start Engine
+	${date}=	Get Current Date	result_format=epoch
+	Log To Console	${date}
+	ADD SVC COMMENT	host_1	service_1	0	user	comment
+	${content}=	Create List	ADD_SVC_COMMENT finished
+	${result}=	Find In Log with Timeout	${logEngine0}	${start}	${content}	60
+	Should Be True	${result}	msg=command argument persistent_flag must be 0 or 1.
```
Comment (bouda1) [inline tests/broker-engine/external-commands.robot:2429]: Also add a call to DEL_SVC_COMMENT with wrong id (-1 for example).

Comment (bouda1) [reply tests/broker-engine/external-commands.robot:2429]: in the centreon_storage database, we have a table `comments` that can help you

File: tests/resources/Engine.py [hunk 4]
```diff
@@ -1348,3 +1350,30 @@ def config_engine_remove_cfg_file(poller: int, fic: str):
     ff = open("{}/config{}/centengine.cfg".format(CONF_DIR, poller), "w")
     ff.writelines(linesearch)
     ff.close()
+
+
+def send_custom_host_notification(hst, notification_option, author, comment):
+    now = int(time.time())
+    cmd = "[{}] SEND_CUSTOM_HOST_NOTIFICATION;{};{};{};{}\n".format(
+        now, hst, notification_option, author, comment)
```
Comment (bouda1) [inline tests/resources/Engine.py:1358]: you can use f""

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: engine/src/commands/commands.cc [hunk 5]
```diff
@@ -214,22 +221,27 @@ int cmd_add_comment(int cmd, time_t entry_time, char* args) {
       (cmd == CMD_ADD_HOST_COMMENT) ? comment::host : comment::service,
       comment::user, temp_host->get_host_id(), service_id, entry_time, user,
       comment_data, persistent, comment::external, false, (time_t)0)};
-  comment::comments.insert({com->get_comment_id(), com});
-
+  uint64_t internal_id = com->get_comment_id();
```
Comment (bouda1) [inline engine/src/commands/commands.cc:224]: here, we are in engine, some let's keep the name comment_id

**Review by bouda1 — CHANGES_REQUESTED** [review]

**Review by jean-christophe81 — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Pull request merged**